step-capstone/src/components/AddEventHotel.js

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect } from 'react'
+import { useState } from 'react'
+import {
+    TextField,
+    Grid,
+    Checkbox,
+    FormControlLabel,
+} from '@material-ui/core';
+import {
+    MuiPickersUtilsProvider,
+    DateTimePicker
+} from '@material-ui/pickers';
+import DateFnsUtils from "@date-io/date-fns";
+
+export default function AddEventHotel(props) {
+    let overwriting = props.data !== undefined;
+
+    // Sets values to previous values if editing, otherwise blank slate
+    const [startDate, handleStartChange] = useState(overwriting ? props.data.startDate : new Date());
+    const [endDate, handleEndChange] = useState(overwriting ? props.data.endDate : new Date());
+    const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
+    const [title, setTitle] = useState(overwriting ? props.data.title : "");
+    const [location, setLocation] = useState(overwriting ? props.data.location : "");
+    const [description, setDescription] = useState(overwriting ? props.data.description : "");
+
+    const handleChecked = (e) => {
+        setChecked(e.target.checked);
+    }
+
+    const handleTitleChange = (e) => {
+        setTitle(e.target.value);
+    }
+
+    const handleLocationChange = (e) => {
+        setLocation(e.target.value);
+    }
+
+    const handleDescriptionChange = (e) => {
+        setDescription(e.target.value);
+    }
+
+    /*
+     * Called once change to hook state is complete. Updates data property in AddForm.
+     */
+    useEffect(() => {
+        console.log(title)
+        props.onDataChange({
+            id: overwriting ? props.data.id : undefined,
+            title: title,
+            type: props.type,
+            startDate: startDate,
+            endDate: endDate,
+            finalized: checked,
+            location: location,
+            description: description
+        })
+
+        //validating input
+        if (title === "") {
+            props.onToggleValidation(false);
+        } else if (checked && location === "") {
+            props.onToggleValidation(false);
+        } else {
+            props.onToggleValidation(true);
+        }
+    }, [startDate, endDate, checked, title, location, description])
+
+    // render error input if no title
+    let renderTitleInput = () => {
+        if (title === "") {
+            return (
+                <TextField
+                    error
+                    helperText="Cannot leave field blank"
+                    id="title"
+                    label={"Add Title"}
+                    defaultValue={title}
+                    fullWidth
+                    onChange={handleTitleChange}
+                />
+            )
```

Comment:
This, together with the above, seems like a lot of code to handle a single validation. Rather than writing basically the same hook for a few different fields, think about whether you can make a common utility so you only do it once.

Reply:
Good point! I'll try to find a way.